analyzer-core/src/main/java/ffb/analyzer/core/queue/Task.java

```diff
@@ -0,0 +1,43 @@
+package ffb.analyzer.core.queue;
+
+import java.util.function.Consumer;
+
+public class Task<T> {
+    public int id;
+
+    private T objectToPerformTaskOn;
+    private Consumer<T> actionToPerform;
+
+
+    public void performTask() {
+        try {
+            actionToPerform.accept(objectToPerformTaskOn);
+        } catch (Exception e) {
+            throw new RuntimeException(); //TODO: Add error logging instead of throwing exception
```

Comment:
I'm thinking we should open a separate issue to implement error logging as a whole.

Reply:
Agreed. We'll eventually have a logger (once we stand up the server portion) that we can access using `@Inject`.